src/components/Tooltip.js

```diff
@@ -0,0 +1,141 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {Animated, Text, View} from 'react-native';
+import Hoverable from './Hoverable';
+import withWindowDimensions, {windowDimensionsPropTypes} from './withWindowDimensions';
+import getTooltipStyles from '../styles/getTooltipStyles';
+
+const propTypes = {
+    // The text to display in the tooltip.
+    text: PropTypes.string.isRequired,
+
+    // Child to wrap with Tooltip. Should be only child, or the tooltip placement will be unpredictable.
+    children: PropTypes.node.isRequired,
+
+    // Props inherited from withWindowDimensions
+    ...windowDimensionsPropTypes,
+};
+
+class Tooltip extends Component {
+    constructor(props) {
+        super(props);
+
+        this.animation = new Animated.Value(0);
+
+        // The wrapper view containing the wrapped content along with the Tooltip itself.
+        this.wrapperView = null;
+        this.wrapperWidth = 0;
+        this.wrapperHeight = 0;
+
+        // The tooltip (popover) itself.
+        this.tooltip = null;
+        this.tooltipWidth = 0;
+        this.tooltipHeight = 0;
+
+        // The distance between the left side of the rendered view and the left side of the window
+        this.xOffset = 0;
+
+        // The distance between the top of the rendered view and the top of the window
+        this.yOffset = 0;
+
+        this.measureWrapperAndGetPosition = this.measureWrapperAndGetPosition.bind(this);
+        this.measureTooltip = this.measureTooltip.bind(this);
+        this.showTooltip = this.showTooltip.bind(this);
+        this.hideTooltip = this.hideTooltip.bind(this);
+    }
+
+    /**
+     * Measure the size and position of the wrapper view.
+     */
+    measureWrapperAndGetPosition() {
+        this.wrapperView.measureInWindow((x, y, width, height) => {
```

Comment:
Might be nice to mention somewhere in the code why we are using `measureInWindow`